data-prepper-plugins/http-source-common/src/test/java/org/opensearch/dataprepper/http/codec/JsonCodecTest.java

```diff
@@ -103,13 +113,49 @@ public void testParseNonJsonFailure() {
     static class JsonArrayWithKnownFirstArgumentsProvider implements ArgumentsProvider {
         @Override
         public Stream<? extends Arguments> provideArguments(ExtensionContext extensionContext) throws Exception {
+            // First test, all chunks smaller than maxSize, but output has 3 lists
```

Comment:
This test is getting quite hard to understand now.

I'm not sure I have a great suggestion. The first suggestion I have is that perhaps you can keep the existing tests, but add a new test case as well?

Reply:
Only two existing tests and that test is already kept as the first test here. The existing test doesn't do proper validations. I can add more comments if that helps.